aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -205,7 +211,7 @@ void FunctionalTensorWrapper::replace_(const Tensor& other) {
   if (dtype() != value_.unsafeGetTensorImpl()->dtype() || layout() != value_.unsafeGetTensorImpl()->layout()) {
     // .to() should not re-entrantly go through functionalization.
     at::AutoDispatchSkipFunctionalize guard;
-    value_ = value_.to(c10::TensorOptions().dtype(dtype()).layout(layout()));
+    value_ = at::_to_copy(value_, c10::TensorOptions().dtype(dtype()).layout(layout()));
```

Comment:
ok this was sneaky, but I noticed that `to.dtype_layout` was showing up directly in some of my FX graphs, which made the reinplace pass unhappy (it doesn't expect to see any composite ops, since e.g. `to()` is a "maybe view" operator).

Why does that happen? autograd has already ran by the time we get to this code (in my crappy implementation of `_functionalize()` in the test suite), so that prevents the composite kernel for `to()` from being called here.

Putting `_to_copy()` into the graph is what we want. I guess this would also get fixed by registering composite kernels directly to the `Python` key though..

Reply:
This looks good, but you should have a comment in the code here